src/engine/parser/error.rs

```diff
@@ -0,0 +1,190 @@
+// This file is part of rust-web/twig
+//
+// For the copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+
+//! Typisation of parser errors.
+
+use std::fmt::{self, Display};
+use error::Error;
+use error::{GeneralizeTo, ErrorCode};
+
+use engine::parser::token::TokenErrorCode;
+use engine::parser::job::{self, cursor};
+use engine::parser::token;
+
+pub type ParserError = Error<ParserErrorCode>;
+pub type NodeError = Error<NodeErrorCode>; // todo move somewhere else??
+
+impl GeneralizeTo<ParserErrorCode> for TokenErrorCode {
+    fn generalize(&self) -> ParserErrorCode { ParserErrorCode::TokenError }
+}
+
+#[allow(dead_code)]
+#[derive(Debug)]
+pub enum ParserErrorCode {
+    Unreachable {
```

Comment:
Yeah, I know. TwigPHP has this "Logic"-Errors - this was my rust-translation. If we ever have some code branch which "*should* be unreachable" but without being 100% certain - this could help debugging. Like possibly race conditions? I don't know. But I agree, we don't neccessarily need it from the start, so I will remove it from the skeleton.